File: solutions/csharp/collatz-conjecture/2/CollatzConjecture.cs
using Xunit.Sdk;

public static class CollatzConjecture
{
    public static int Steps(int number)
    {
        ArgumentOutOfRangeException.ThrowIfNegativeOrZero(number);

        var steps = 0;
        var result = number;

        while (result != 1)
        {
            result = AdjustResult(result);
            steps++;
        }

        return steps;
    }

    private static int AdjustResult(int result)
    {
        return result % 2 == 0 ? result / 2 : result * 3 + 1;
    }
}

File: solutions/ruby/collatz-conjecture/2/collatz_conjecture.rb
class CollatzConjecture
  def self.steps(n)
    raise ArgumentError unless n > 0

    count = 0
    while n != 1
      count += 1

      if n % 2 == 0 then
        n /= 2
      else
        n = (n * 3) + 1
      end
    end

    count
  end
end